File: scripts/036/yamaguchi.rb
#Problem 36 †
#585 = 10010010012 (2進) は10進でも2進でも回文数である.
#100万未満で10進でも2進でも回文数になるような数の総和を求めよ.
#(注: 先頭に0を含めて回文にすることは許されない.)
max = 1_000_000
sum = 0
1.upto(max) do |num|
  str_num = num.to_s
  next if str_num != str_num.reverse
  bin_num = num.to_s(2)
  sum += num if bin_num == bin_num.reverse
end
p sum

File: scripts/036/yamaguchi.js
String.prototype.reverse = function (){
    var t = "";
    for(var i = this.length; i > 0; i--) t = t + this.substring(i - 1, i);
    return t;
}
var max = 1000000;
var sum = 0;
var i = 1;
var str_num = null;
for(; i <= max; ++i){
    str_num = i.toString();
    if(str_num !== str_num.reverse()){
        continue;
    }

    bin_num = i.toString(2);
    if(bin_num === bin_num.reverse()){
        sum += i;
    }
}
console.log(sum);